app/assets/javascripts/cart.js.coffee

```diff
@@ -45,6 +42,8 @@ $(document).ready ->
       $('#modal_place').modal('show')
       $('#modal_place .yes-button').data('srid', srid)
       $('#modal_place .yes-button').data('service-id', id)
+      $('#modal_place .yes-button').on 'click', (e) ->
```

Comment:
@kyle-glick When I was debugging this file, the variables set after $(document).on 'click', '.remove-service', -> were not being reset.  So if you deleted one line_item, everything would work fine.  However, if you deleted a second one.  It would try to execute the code with the previous line item id and then it would also execute the code for the line item you just removed (I think that is what you were describing above- multiple clicks). Changing it to this selector on click format solved the problem for me.